File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
import sys

input = sys.stdin.readline

original = input().strip()
boom = input().strip()
stack = []
boom_size = len(boom)
last_char = boom[-1]  # 폭발 문자열의 마지막 문자

for char in original:
    stack.append(char)
    # 스택의 마지막 문자가 폭발 문자열의 마지막 문자와 같을 때만 검사
    if char == last_char and ''.join(stack[-boom_size:]) == boom:
        del stack[-boom_size:]  # 슬라이싱 삭제

# 결과 출력
if not stack:
    print("FRULA")
else:
    print(''.join(stack))

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.kt
package org.example

fun main(){
    val br = System.`in`.bufferedReader()
    var stack = StringBuilder()
    val original = br.readLine()
    val boom = br.readLine()
    val boomSize = boom.length
    val boomLast = boom.last()

    for(char in original){
        stack.append(char)
        if(char == boomLast && stack.length >=boomSize){
            if(stack.substring(stack.length-boomSize)==boom){
                stack.delete(stack.length - boomSize, stack.length) // 슬라이싱 삭제
            }
            
        }
    }

    if (stack.isEmpty()) {
        println("FRULA")
    } else {
        println(stack.toString())
    }
}